File: Euler23.py
import math


def factor(x):
    divisors = {1}
    largest = int(math.sqrt(x))
    if largest * largest != x:
        largest += 1
    else:
        divisors.add(largest)
    for i in range(2, largest):
        if x % i == 0:
            divisors.add(i)
            divisors.add(x/i)
    return(divisors)


def abundant(n):
    if sum(factor(n)) > n:
        return True
    return False


rng = {x: abundant(x) for x in list(range(1, 28126))}
abdntNums = [item for item in rng if rng[item] == True]
print("Abundant numbers: ")
print(abdntNums)


def abundantSum(num):
    i = 0
    while abdntNums[i] < num//2+1:
        if rng[num - abdntNums[i]] == True:
            return True
        i += 1
    return False


nonAbSumNums = [i for i in range(1, 28125) if abundantSum(i) == False]
print("Non abundant sum numbers: ")
print(nonAbSumNums)

sum = 0
for val in nonAbSumNums:
    sum += val
print(sum)

File: Euler23.js
var math = require('mathjs');

var abNums = [];
var nonAbSumNums = [];

function factor(x) {
  var divisors = [];

  var largest = int(math.sqrt(x));
  if(largest * largest != x){
    largest += 1;
  }

  for(int i = 1; i <= largest; i++){
    if(x%i = 0){
      divisors.push(i);
      divisors.push(x/i);
    }
  }

  return divisors;
}

function abundantCheck(n) {
  var total = 0;
  for(var i in arr){
    total += n[i];
  }

  if(total > n){
    abNums.push(n);
  }
}

function abSumCheck(arg) {
  var total = 0;
    for(var i in abNums){

    }
}

for(var i=1; i < 28124; i++){
  abundantCheck(i);
}

for(var i=23; i < 28123; i++){
  abSumCheck(i);
}
